diesel/src/query_dsl/save_changes_dsl.rs

```diff
@@ -69,8 +69,10 @@ where
 pub trait SaveChangesDsl<Conn> {
     /// Sugar for types which implement both `AsChangeset` and `Identifiable`
     ///
+    /// On backends which support the `RETURNING` keyword,
     /// `foo.save_changes(&conn)` is equivalent to
-    /// `update(foo::table().find(foo.id())).set(&foo).get_result(&conn)`
+    /// `update(&foo).set(&foo).get_result(&conn)`.
+    /// On other backends, two queries will be executed.
```

Comment:
Oh, these two queries are not executed in a transaction. Maybe they should be.

Reply:
It only executes one query that modifies data (the second query is a select). A transaction is only really needed if you're executing two or more queries that modify things, which you'd like to have be atomic. If that were the case, you would need an outer transaction regardless of backend.

Reply:
I was more concerned with the data being changed by another user/connection while you're waiting for the second query :)

Reply:
This is a post 1.0 thing either way. I'll open an issue.